src/devices/bus/isa/areplay.cpp

```diff
@@ -0,0 +1,614 @@
+
+        TODO: 
+	(*) unblock flash upgrades, currently intelflash doesn't cooperate
+
+	- XMS / EMS memory not really tested 
+
+	- compatibility with * official * 5.0 binaries (if they exist)
+
+	- improve banking; add missing / secret bits.
+	  Bit 7 in read register seems to be added in later V5 hardware.
+
+          No 'magic bits' to disable ROM/RAM or cart were found. Bit 5
+          in first write register is set when freezer menu is active.
+*********************************************************************************/
+#include "emu.h"
+#include "areplay.h"
+
+//**************************************************************************
+//  DEVICE DEFINITIONS
+//**************************************************************************
+DEFINE_DEVICE_TYPE(ISA8_AREPLAY, isa8_areplay_device, "areplay", "Action Replay for DOS - DATEL UK 1994")
+
+//-------------------------------------------------
+//  ROM( action replay )  V4.8 or V5.0
+//-------------------------------------------------
+
+// - official V4.8 firmware = CRC(6c4a64ce)
+// - encoded V5.0 beta firmware  = CRC(0ff4d5ef)
+ROM_START( areplay )
+	ROM_REGION(0x20000,"v48rom.rom",0)
+	ROM_LOAD( "v48rom.rom", 0x0000, 0x20000, CRC(6c4a64ce) SHA1(fac68327a4c8b3cb03def2a69e5930c845a09047) )
+
+	ROM_REGION(0x40000,"v50rom.rom",0)
+	ROM_LOAD( "v50rom.rom", 0x0000, 0x20000, CRC(0ff4d5ef) SHA1(eb4cd4f71ebf6d604b8e7924141389585b92ff5e)  )
+ROM_END
+
+//-------------------------------------------------
+//  rom_region - device-specific ROM region
+//-------------------------------------------------
+const tiny_rom_entry *isa8_areplay_device::device_rom_region() const
+{
+	return ROM_NAME( areplay );
+}
+
+//-------------------------------------------------
+//  device_add_mconfig - add device configuration
+//-------------------------------------------------
+MACHINE_CONFIG_START(isa8_areplay_device::device_add_mconfig)
+	MCFG_TIMER_DRIVER_ADD_PERIODIC("heartbeat", isa8_areplay_device, heartbeat_timer, attotime::from_hz(18))
+MACHINE_CONFIG_END
+//**************************************************************************
+
+//-------------------------------------------------
+//  isa8_areplay_device - constructor
+//-------------------------------------------------
+isa8_areplay_device::isa8_areplay_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+	device_t(mconfig, ISA8_AREPLAY, tag, owner, clock),
+	device_isa8_card_interface(mconfig, *this)
+{
+}
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+void isa8_areplay_device::device_start()
+{
+	set_isa_device();
+
+	one_shot_timer = timer_alloc(0);
+}
+
+//-------------------------------------------------
+//  device_reset - device-specific reset
+//-------------------------------------------------
+void isa8_areplay_device::device_reset()
+{
```

Comment:
As I commented on a previous PR, you need to keep a flag indicating that you've installed handlers and not reinstall them on subsequent resets.  Otherwise if someone changes config you end up mapped in multiple places after a soft reset.  Due to various limitations in the memory system, this is the easiest way to do things at the moment - config changes need a hard reset.